core/interfaces/branch.py

```diff
@@ -502,18 +502,23 @@ def run(self, edit):
 
     def run_async(self):
         self.interface = ui.get_interface(self.view.id())
-        current_branch_name = self.get_current_branch_name()
 
+        branches = self.get_selected_branches()
+        branches_strings = self.create_branches_strs(branches)
+        self.merge(branches_strings) if not len(branches_strings) == 0 else None
+        util.view.refresh_gitsavvy(self.view)
+
+    def get_selected_branches(self):
+        current_branch_name = self.get_current_branch_name()
         branches = set()
         for sel in self.view.sel():
-            for name in self.get_selected_branches(sel):
+            for name in self.get_selected_branch(sel):
                 if name and not name == current_branch_name:
                     branches.add(name)
 
-        self.merge(branches) if not len(branches) == 0 else None
-        util.view.refresh_gitsavvy(self.view)
+        return branches
 
-    def get_selected_branches(self, sel):
+    def get_selected_branch(self, sel):
```

Comment:
I kind of like that. And since we're in an interface already, we can have a shortcut property such as `self.interface`. but the again, do we have any other view that allows selecting branches other than branch interface?

Personally, I just don't like to read a "jumping flow" like that. Which is basically one code unit broken up and sorted reversed. The total row count is not that significant as well.
But, anyway this is totally personal preference. Not worth discussing any more 😄 
(p.s comment moved to proper review context)